airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -288,26 +288,14 @@ def _read_full_refresh(
         configured_stream: ConfiguredAirbyteStream,
         internal_config: InternalConfig,
     ) -> Iterator[AirbyteMessage]:
-        slices = stream_instance.stream_slices(sync_mode=SyncMode.full_refresh, cursor_field=configured_stream.cursor_field)
-        logger.debug(
-            f"Processing stream slices for {configured_stream.stream.name} (sync_mode: full_refresh)", extra={"stream_slices": slices}
-        )
         total_records_counter = 0
-        for _slice in slices:
-            if self._slice_logger.should_log_slice_message(logger):
-                yield self._slice_logger.create_slice_log_message(_slice)
-            record_data_or_messages = stream_instance.read_records(
-                stream_slice=_slice,
-                sync_mode=SyncMode.full_refresh,
-                cursor_field=configured_stream.cursor_field,
-            )
-            for record_data_or_message in record_data_or_messages:
-                message = self._get_message(record_data_or_message, stream_instance)
-                yield message
-                if message.type == MessageType.RECORD:
-                    total_records_counter += 1
-                    if internal_config.is_limit_reached(total_records_counter):
-                        return
+        for record_data_or_message in stream_instance.read_full_refresh(configured_stream.cursor_field, logger, self._slice_logger):
```

Comment:
TODO: write unit tests

Reply:
@girarda do we really need to pass a logger instance here?

Reply:
The assumption is no, but we each source is free to do whatever they want. It is possible some of the use the logger passed as a parameter instead of `self.logger`. There's no way to guarantee none of them are doing this since we do not control custom connectors.

The way forward will be to use the `AbstractStream`'s interface, which does not pass a logger, but actually deprecating the "legacy Stream" interface won't happen any time soon. [The StreamFacade acts as a bridge between the two interface (and does not pass the logger argument to the AbstractStream)](https://github.com/airbytehq/airbyte/pull/30111/files#diff-2bd61ee4aff22d10cae021d182dc9622a89d426eae2779ecf2be27ddfea88074R117-R123)